compiler/src/main/java/com/rejasupotaro/android/kvs/internal/SchemaWriter.java

```diff
@@ -21,129 +25,144 @@ public SchemaWriter(SchemaModel model) {
 
     public void write(Filer filer) {
         try {
-            StringBuilder fqcn = new StringBuilder();
-            fqcn.append(model.getPackageName())
-                    .append(".")
-                    .append(model.getClassName());
-
-            JavaFileObject sourceFile = filer.createSourceFile(fqcn.toString(), model.getElement());
-            JavaWriter writer = new JavaWriter(sourceFile.openWriter());
-
-            writer.emitPackage(model.getPackageName());
-            writeImports(writer);
-            writer.beginType(model.getClassName(), "class", EnumSet.of(Modifier.PUBLIC, Modifier.FINAL), model.getOriginalClassName());
+            TypeSpec.Builder classBuilder = TypeSpec.classBuilder(model.getClassName());
+            classBuilder.addModifiers(Modifier.PUBLIC, Modifier.FINAL);
+            ClassName superClassName = ClassName.get(model.getPackageName(), model.getOriginalClassName());
+            classBuilder.superclass(superClassName);
+
+            List<FieldSpec> fieldSpecs = createFields();
```

Comment:
Unfortunately, we can't use `addFields` because these interfaces are not released yet :/

- [Add Collection utilities on *Spec builders. · square/javapoet@b507426](https://github.com/square/javapoet/commit/b5074260)
- [compare/javapoet-1.0.0...master](https://github.com/square/javapoet/compare/javapoet-1.0.0...master)